tests/e2e/basic_test.go

```diff
@@ -20,34 +20,54 @@
 package e2e
 
 import (
-	"log"
-	"os"
-
 	. "gopkg.in/check.v1"
 
+	"github.com/vmware/docker-volume-vsphere/tests/utils/admincli"
 	"github.com/vmware/docker-volume-vsphere/tests/utils/dockercli"
 	"github.com/vmware/docker-volume-vsphere/tests/utils/inputparams"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/misc"
 	"github.com/vmware/docker-volume-vsphere/tests/utils/verification"
 )
 
+const (
+	testName = "basic_test"
+	vmgroup  = "T1"
+)
+
 type BasicTestSuite struct {
-	esxName       string
-	volumeName    string
+	config        *inputparams.TestConfig
+	esx           string
+	vm1           string
+	vm2           string
+	vm1Name       string
+	vm2Name       string
+	volName1      string
+	volName2      string
 	containerName string
 }
 
-func (s *BasicTestSuite) SetUpTest(c *C) {
-	s.esxName = os.Getenv("ESX")
-	s.volumeName = inputparams.GetVolumeNameWithTimeStamp("basic_test")
-	s.containerName = inputparams.GetContainerNameWithTimeStamp("basic_test")
+func (s *BasicTestSuite) SetUpSuite(c *C) {
+	s.config = inputparams.GetTestConfig()
+	if s.config == nil {
+		c.Skip("Unable to retrieve test config, skipping basic tests")
+	}
+
+	s.esx = s.config.EsxHost
+	s.vm1 = s.config.DockerHosts[0]
+	s.vm2 = s.config.DockerHosts[1]
+	s.vm1Name = s.config.DockerHostNames[0]
+	s.vm2Name = s.config.DockerHostNames[1]
+	s.volName1 = inputparams.GetUniqueVolumeName(testName)
+	s.volName2 = inputparams.GetUniqueVolumeName(testName)
+	s.containerName = inputparams.GetContainerNameWithTimeStamp(testName)
```

Comment:
Personally I don't really have much concern about this. We have already introduced common TestConfig, which I feel is sufficient. Here I introduced these local variables just for convenience because I personally prefer short names locally. Otherwise I would just use s.config.XXX.